ffi/src/winscard/scard.rs

```diff
@@ -64,20 +81,24 @@ pub unsafe extern "system" fn SCardConnectA(
     check_null!(pdw_active_protocol);
 
     let reader_name = try_execute!(
+        // SAFETY: The `sz_reader` parameter is not null (checked above).
         unsafe { CStr::from_ptr(sz_reader as *const i8) }.to_str(),
         ErrorKind::InvalidParameter
     );
 
-    try_execute!(unsafe {
-        connect(
-            context,
-            reader_name,
-            dw_share_mode,
-            dw_preferred_protocols,
-            ph_card,
-            pdw_active_protocol,
-        )
-    });
+    try_execute!(
+        // SAFETY: All parameters are validated and/or type checked.
```

Comment:
note: This is not really a good way of proving that the usage is safe as it’s almost like not explaining at all why it’s safe. If it’s really trivial, I would say it’s fine though.

Reply:
I've improved it a little bit: [`7b6cb00` (#250) `ffi/src/winscard/scard.rs`](https://github.com/Devolutions/sspi-rs/pull/250/commits/7b6cb00e1b2110172d2de88e2374daa22cc8fc32#diff-0426fa6cf83bb8cdd88c0e5aba030b21d7e6b12d26e7fc2d514f1d000cfb223a)